File: 14. Longest Common Prefix/14. Longest Common Prefix.js
var longestCommonPrefix = function(strs){
    if (strs.length == 0) return ""

    for (let i = 0; i < strs[0].length; i++){
        for (str of strs){
            if (i >= str.length || strs[0][i] != str[i]){
                return strs[0].slice(0, i)
            }
        }
    }

    return strs[0]
}

File: 14. Longest Common Prefix/14. Longest Common Prefix.py
class Solution : 
    def longestCommonPrefix(self, strs):
        if not strs : 
            return ""
        
        for i in range(len(strs[0])):
            for s in strs:
                if i > len(s) or strs[0][i] != s[i]:
                    return strs[0][:i]
        
        return strs[0]